### ISSUE TITLE
Various accessibility issues 

### ISSUE THREAD

**Issue by VinoManoTR** [scalar/scalar#4105]
### What happens?

1. Screen reader doesn't announce base URL as a read-only input field. 
![image](https://github.com/user-attachments/assets/1c73be01-4e23-4602-bfa4-0c0315af0548)

2. Visual labels announced by the screen reader should be should be the same values that are displayed: Eg. “Server” and “Bearer authentication (Bearerauth)
![image](https://github.com/user-attachments/assets/bc179335-6cd8-420b-a480-df35c09a298e)

3. Navigation bar should be announced as a "table of contents"
![image](https://github.com/user-attachments/assets/0232b196-f7a8-4880-970f-3b9429e26a98)

4. The screen reader should announce the "Show password button", after activating this element the screen reader should announce "Hide password button". The role should be set to "button" instead of "checkbox" and this means the button doesn't need the "checked" and "collapsed" states
![image](https://github.com/user-attachments/assets/a322b095-d708-4855-ba9a-a1d6e2d28175)

5. The Show Child Attributes expanded are missing aria-control attribute for the button, a role for the expanded region, and the elements are not implemented as a list.
![image](https://github.com/user-attachments/assets/d7cbdd71-ffeb-44dc-949e-80056ebd98f4)

6. Using the keyboard, a user can't focus on the scrollable section of the code snippet. Focus moves from "Copy to clipboard" to "Test Request" button
![image](https://github.com/user-attachments/assets/321261d6-838f-4b12-bcec-c44efd819aba)

7. Screen reader does not announce status messages after activating buttons '# Copy link to' and 'Copy Content'.

8. On zoom 400% content endpoints and code snippets are not fully visible. Only headers are visible
![image](https://github.com/user-attachments/assets/b1fbc58e-8b4c-4b45-b473-c545eaf16697)

### What did you expect to happen?

All active elements on the test request page should be keyboard accessible.
When the keyboard focus is on an element, there should be a visual indicator of the focus (usually a box highlighting the element).
The screen reader should be able to make an announcement of each element that includes context for the visual label.
